Use abtests from Calypso

bsessions85 opened the pull request:
#### Changes proposed in this Pull Request

* This changes e2e tests from using default config for abtests to using active-tests.js

#### Testing instructions
* Ensure tests pass in CircleCI
* Run locally an make sure that after `setABTestControlGroupsInLocalStorage` is run, all tests and defaults from active-tests.js are set properly in local storage

Before:
![image](https://user-images.githubusercontent.com/31110506/55001357-080ae380-4f9a-11e9-8fff-96ac18c4677e.png)

After:
![image](https://user-images.githubusercontent.com/31110506/55001138-903cb900-4f99-11e9-8e10-fb40e8daa57b.png)

Commit:
Use abtests from Calypso

Commit:
Use abtests from Calypso

jsnajdr committed:
Domains: remove unused .icann-verification styles

The stylesheet was duplicated/migrated to `email-verification` in #19577 but never removed.

jsnajdr committed:
Domains: migrate .email-verification CSS styles to webpack

Also remove global override for `.notice__text` (added in #15880)

jsnajdr committed:
Domains: shorten the import path of IcannVerificationCard component

Commit:
cleanup a bit

Commit:
cleanup a bit

Commit:
Merge branch 'master' into try/use-calypso-abtests

blowery commented on test/e2e/lib/async-base-container.js:
```diff
@@ -108,14 +109,23 @@ export default class AsyncBaseContainer {
+		let expectedABTestValue = [];
+		const abTestList = abTests.default;
+
+		Object.keys( abTestList ).forEach( function( test ) {
```
this could use `.map` instead and return directly into `expectedABTestValue`.

```
const expectedABTestValue = Object.keys( abTestList ).map( test => `"${test}_${abTestList[ test ].datestamp}:${abTestList[ test ].defaultVariation}"` )
```

bsessions85 replied:
Great, thanks for the suggestion!

Commit:
using map instead

Commit:
using map instead

Stojdza committed:
Add NODE_CONFIG_ENV instead NODE_ENV (#31632)

* Add NODE_CONFIG_ENV instead NODE_ENV

* update package-lock

* Update warnings

* Update run-wrapper.sh

* Update circle-ci-branch-update.sh

* Update .circleci/config.yml

* Revert "Update run-wrapper.sh"

* Revert "Revert "Update run-wrapper.sh""

This reverts commit e308b10

* Update docs

scruffian committed:
Change the way we handle positioning when a guided tour will overlap the right side of the screen

sirreal committed:
Jetpack blocks: Remove Jetpack blocks artifact fetching (#31760)

This was used for an automated synchronization process that is disabled
and will not be restored.

Remove the fetch code and the integration test.

sirreal committed:
Jetpack blocks: Remove CircleCI build (#31757)

No longer in use.

sirreal committed:
SDK: Failed builds should exit with error status (#30019)

Stojdza committed:
[e2e] Upgrade config (#31764)

* Upgrade config

* update package-lock.json

Stojdza reviewed (approved):
Looks great! Thanks @bsessions85!

aidvu committed:
Domains: Search not showing map/transfer option

We didn't limit the scope of a prop, so if it got stored
into local storage, it would still be available outside
of NUX, thus hiding the option from regular domain search.

cbauerman committed:
Fix Gsuite TOS Link (#31738)

cbauerman committed:
Switch Email and Domain Management Email Paths (#31703)

ramonjd committed:
Inline Popover: Fix/close onboarding welcome prompt (#31745)

* Ensuring the welcome prompt goes away whenever the popover is given the instruction to hide.

* added reducer tests

southp committed:
Add: the data layer handlers & actions for interacting with `/verticals` endpoint (#31714)

* Add data-layer handlers and the actions for interacting with
`/verticals` v2 API.

* Move the verticals data actions into its own `signup/verticals` folder,
since it doesn't really belong to any signup steps.

* Use `convertToCamelCase()` as the `fromApi()` function for `/verticals`
data-layer handler.

* Update the path for importing `setVerticals`

* Add the test suite for signup/verticals/actions

* Add the test suite for /signup/verticals data-layer handlers.

southp committed:
Add: verticals substate tree (#31716)

* Add the new `verticals` reducer

* Add `getVerticals` selector.

* Add a test suite for `signup/verticals/selectors`

* Add a test suite for the signup/verticals/reducer

southp committed:
Add: <QueryVerticals/> component (#31718)

* Implement <QueryVerticals/> component

* query-signup-verticals => query-verticals

* Make use of `componentDidUpdate()` life cycle method to make sure
`QueryVerticals` component is fetching per updates.

* Give searchTerm an empty string as the default

* Add a test suite for QueryVerticals component

southp committed:
Wiring the new `verticals` state tree to the related components (#31746)

* Plumb the underlying `signup.verticals` state tree onto
`SiteVerticalsSuggestionSearch` component

* Draft: making the site vertical data a derived data from selector,
rather than a vanilla state tree.

* Add a `isNavigating` parameter to the `onChange()` callback of the
component `SuggestionSearch`

* Store the candidate vertical list in a state so that a user can navigate
through it without updating the list, since the props `verticals` is
changing depending on the `searchValue` props.

* Derive `isVerticalSearchPending` from `verticals` props.

* Tweak the definition of isVerticalSearchPending.

* Move some selectors down and add a TODO comment to note down the future
polishing plan.

* Update the test suite of site-vertical/selectors

* Update the `about` step to accommodate the related changes of
<SiteVerticalsSuggestionSearch/>

* Fix the <SiteVerticalsSuggestionSearch /> component tests and remove
unused props.

* More TODOs

* Add the missing constants.js

* Remove the unused `lastUpdated`

* Use `getSiteVerticalName()` for getting the site vertical name.

* Fix a type

jsnajdr committed:
UpgradeNudgeExpanded: migrate CSS to webpack (#31758)

* UpgradeNudgeExpanded: remove Media/VideoPress styling

It was used in an A/B test that didn't succeed (#8403).

* UpgradeNudgeExpanded: migrate CSS to webpack

jsnajdr committed:
Domains: migrate ICANN/email verification CSS to webpack (#31751)

* Domains: remove unused .icann-verification styles

The stylesheet was duplicated/migrated to `email-verification` in #19577 but never removed.

* Domains: migrate .email-verification CSS styles to webpack

Also remove global override for `.notice__text` (added in #15880)

* Domains: shorten the import path of IcannVerificationCard component

brbrr committed:
E2E tests: Fix Jetpack connect (and others) tests  (#31668)

* Move e2e-related ingores to e2e folder

* Some code clean-up

* Update jetpack connect tests to reflect recent onboarding changes

* Add UserType JPO step

* More fixes here and there

* Dismiss a stats popup for invite test

sirreal committed:
Make SDK Gutenberg generic (#31723)

The SDK Gutenberg tool had become overly specific to build the Jetpack
Gutenberg extensions. Said extensions no longer depend on the SDK tool,
instead relying on their own specific build tool.

Remove the changes that were geared towards the Jetpack builds.

keoshi committed:
My Plan: reorder plan feature cards to highlight focus on security and performance (#31749)

* Tighted up current plan header

* Add class to plan header if is free plan

* Add conditional styles for free plan

* Reorder feature cards

* Fix plan icon vertical spacing in paid plans

* Bump the minimum required screen size to show the button on the right

* Hide plan icon on small screens

tyxla committed:
Components: Migrate HrWithText CSS to Webpack (#31794)

* Components: Migrate HrWithText CSS to Webpack

* Remove prettier pragma

sirreal committed:
Framework: Make JSON pretty too (#31554)

* Prettier all json
* Add some json to eslintignore

tyxla committed:
Blocks: Migrate EditGravatar CSS to Webpack (#31795)

* Blocks: Migrate EditGravatar CSS to Webpack

* Remove prettier pragma

keoshi committed:
Jetpack Onboarding: add accent color, clean up styles (#31625)

* Onboarding: add overriding color var for Jetpack flow

* Replace JP color for var

* Remove unnecessary overriding rules

* Move color overrides to the top, add more

* Delete unused styles

* Update color variables name

* Add overrides for checkout page header

sirreal committed:
Jetpack blocks: Remove 💥 (#31759)

* Remove jetpack-blocks package
* Remove babel override
* Remove from CODEOWNERS
* Remove now-obsolete mapbox-gl dep
* Remove unused copy-webpack-plugin
* Remove Jetpack blocks mention in client/gutenberg/extensions/README.md

tyxla committed:
Layout: Exclude atomic sites from JP logic (#31797)

Copons committed:
Block Editor: Update the featured image preview after editing the image file (#31761)

jsnajdr committed:
Domains: remove privacy-protection-card styles (#31796)

scruffian committed:
Remove the rect.width parameter when xAboveBelow is called, as its no longer used

matthusby committed:
Blogger Plan: Allow more then 1 `.blog` domain on the blogger plan. (#30999)

scruffian committed:
Merge pull request #31763 from Automattic/update/guided-tours-position

Change the way we handle positioning for guided tours

Commit:
Handling test overrides

Commit:
Handling test overrides

michaeldcain committed:
Post/Page List: add section headers with 'add new' buttons (#31783)

* Add section header with 'add' button to PostTypeList.

* Add section header with 'add' button to page list.

Commit:
needed a null check

Commit:
needed a null check

Commit:
use wrapper in async base container

Commit:
use wrapper in async base container

bsessions85 commented:
Did some refactoring to make the test override functionality work.

Aurorum committed:
WooCommerce Signup: Add Responsiveness (#31049)

* WooCommerce Signup: Add Responsiveness

* Comply Breakpoint with Guidelines

* Prevent the link and button looking too squashed

* Comply with guidelines by building for mobile

* Whitespace

cphilleo committed:
Prevent Jetpack sites displaying WordAds Instant Activation Toggle (#31743)

* Prevent Jetpack sites from showing WordAds instant activation toggle

* Use the correct plan name in the instant activation toggle

Stojdza reviewed (commented):
Looks good 👌 Failures are unrelated.

Commit:
Merge branch 'try/use-calypso-abtests' of github.com:Automattic/wp-calypso into try/use-calypso-abtests
